rswag-specs/lib/rswag/specs/response_validator.rb

```diff
@@ -73,21 +73,23 @@ def validate_body!(metadata, swagger_doc, body)
       end
 
       def validation_options_from(metadata)
+        is_strict = @config.openapi_strict_schema_validation
+
         if metadata.key?(:swagger_strict_schema_validation)
           Rswag::Specs.deprecator.warn('Rswag::Specs: WARNING: This option will be removed in v3.0 please use openapi_all_properties_required and openapi_no_additional_properties set to true')
           is_strict = !!metadata[:swagger_strict_schema_validation]
-        else
```

Comment:
The unconditional else statement here is the source of the bug where we would always receive the deprecation warning regardless of whether the deprecated settings were used or not.